WriteBufferingAndExceptionHandlerTest is flaky

https://travis-ci.org/grpc/grpc-java/jobs/518029993

```
> Task :grpc-netty:test
io.grpc.netty.WriteBufferingAndExceptionHandlerTest > channelCloseFailuresPropagated FAILED
    io.netty.channel.ChannelException: address already in use by: [id: 0x285e4dff, L:local:local]
344 tests completed, 1 failed, 1 skipped
> Task :grpc-netty:test FAILED
FAILURE: Build failed with an exception.
```

Looks like the `local` address is not being cleaned up fast enough

--- a/netty/src/test/java/io/grpc/netty/WriteBufferingAndExceptionHandlerTest.java
+++ b/netty/src/test/java/io/grpc/netty/WriteBufferingAndExceptionHandlerTest.java
@@ -41,6 +41,7 @@ import io.netty.channel.local.LocalAddress;
 import io.netty.channel.local.LocalChannel;
 import io.netty.channel.local.LocalServerChannel;
 import java.net.ConnectException;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicReference;
@@ -69,14 +70,14 @@ public class WriteBufferingAndExceptionHandlerTest {
   private Channel server;
 
   @After
-  public void tearDown() {
+  public void tearDown() throws InterruptedException {
     if (server != null) {
-      server.close();
+      server.close().sync();
     }
     if (chan != null) {
-      chan.close();
+      chan.close().sync();
     }
-    group.shutdownGracefully();
+    group.shutdownGracefully(0, 10, TimeUnit.SECONDS).sync();
   }
 
   @Test

